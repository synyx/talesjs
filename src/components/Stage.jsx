import React from "react";
import PropTypes from "prop-types";

export default class Stage extends React.Component {
  static propTypes = {
    image: PropTypes.object,
  };

  render() {
    const { image } = this.props;
    return <div>{image ? <img src={image.src} /> : null}</div>;
  }
}
