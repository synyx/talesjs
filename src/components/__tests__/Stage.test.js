import React from "react";
import Stage from "../Stage";
import { renderShallow } from "../../../test";

describe("Stage", () => {
  it("renders without image", () => {
    const tree = renderShallow(<Stage />);
    expect(tree).toMatchSnapshot();
  });

  it("renders with image", () => {
    const image = { src: "awesome-image-source" };
    const tree = renderShallow(<Stage image={image} />);
    expect(tree).toMatchSnapshot();
  });
});
