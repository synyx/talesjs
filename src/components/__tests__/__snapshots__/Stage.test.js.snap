// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage renders with image 1`] = `
<div>
  <img
    src="awesome-image-source"
  />
</div>
`;

exports[`Stage renders without image 1`] = `<div />`;
